docs/datamodel/links.rst

```diff
@@ -475,11 +475,141 @@ in these cases due to better ergonomics of selecting, updating, and even
 casting into :eql:type:`json` when keeping all data in the same place rather
 than spreading it across link and object properties.
 
+
+Inserting and updating link properties
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To add a link with a link property, add the link property to a shape on the
+linked object being added. Be sure to prepend the link property's name with
+``@``.
+
+.. code-block:: edgeql
+
+    insert Person {
+      name := "Bob",
+      family_members := (
+        select detached Person {
+          @relationship := "sister"
+        }
+        filter .name = "Alice"
+      )
+    }
+
```

Comment:
I noticed when playing around with this that link properties cannot be made required in a schema, I wonder where would be a good place to add that note. I was thinking maybe a sample query output here showing the relationship between Bob and Alice, then point out to the reader that the relationship shows up as {} if not indicated as it is not required...and indeed link properties cannot be made required so this schema will not work:

```
type Person {
  name: str;
  multi family_members: Person {
    required relationship: str; # Won't work
  }
}
```

But maybe some other part of the page or via a different introduction could work. Or even just skip all that and add a quick "link properties cannot be made required" here:

"They cannot contain links to other objects, and cannot be `required`."

Reply:
Didn't note this here, but I did add it in two other places as noted in previous comment.